runtime/vm/UpcallVMHelpers.cpp

```diff
@@ -373,7 +375,7 @@ native2InterpJavaUpcallImpl(J9UpcallMetaData *data, void *argsListPointer)
 	}
 
 	if (!throwOOM) {
-		returnStorage = currentThread->returnValue;
+		returnStorage = (U_64)currentThread->returnValue;
```

Comment:
This needs to read 64 bits even if `returnValue` is narrower than that. The line should be
```
		/* Read returnStorage from returnValue (and returnValue2 on 32-bit platforms). */
		returnStorage = *(U_64 *)&currentThread->returnValue;
```

Reply:
Updated as suggested above.